packages/ui/src/IconButton.tsx

```diff
@@ -1,183 +1,155 @@
-      ...IndicatorNumPosition[indicatorStyle.position],
-    };
+  const {backgroundColor, color} = getStyles();
 
-    // TODO: remove indicator render function
-    function renderIndicator(): React.ReactElement | null {
-      if (indicator && indicatorNumber && indicatorNumber > 0) {
-        return (
-          <View style={numberIndicatorProps as ViewStyle}>
-            <Box
-              alignItems="center"
-              color={indicatorStyle.color}
-              dangerouslySetInlineStyle={{
-                __style: {
-                  padding: indicatorNumber && indicatorNumber > 9 ? 2 : 0,
-                },
-              }}
-              justifyContent="center"
-              minHeight={15}
-              minWidth={15}
-              rounding="md"
-            >
-              <Text bold color="inverted" size="sm">
-                {indicatorNumber}
-              </Text>
-            </Box>
-          </View>
-        );
-      } else if (indicator) {
-        return (
-          <View style={indicatorPosition as ViewStyle}>
-            {/* TODO: indicator probably needs to be the direct icon, not Ferns UI Icon */}
-            <Icon iconName="circle" size="sm" />
-          </View>
-        );
-      } else {
-        return null;
+  return (
+    <Pressable
+      accessibilityHint={
+        withConfirmation
+          ? `Opens a confirmation dialog to confirm ${accessibilityLabel}`
+          : `Press to perform ${accessibilityLabel} action`
       }
-    }
-
-    const renderConfirmation = () => {
-      return (
-        <Modal
-          primaryButtonOnClick={() => {
-            onClick();
-            setShowConfirmation(false);
-          }}
-          primaryButtonText="Confirm"
-          secondaryButtonOnClick={(): void => setShowConfirmation(false)}
-          secondaryButtonText="Cancel"
-          size="sm"
+      accessibilityLabel={accessibilityLabel}
+      accessibilityRole="button"
+      disabled={loading}
+      style={{
+        alignItems: "center",
+        backgroundColor,
+        borderRadius: theme.radius.rounded as any,
+        justifyContent: "center",
+        height: 32,
+        width: 32,
+      }}
+      testID={testID}
+      onPress={debounce(
+        async () => {
+          await Unifier.utils.haptic();
+          setLoading(true);
+          try {
+            if (withConfirmation && !showConfirmation) {
+              setShowConfirmation(true);
+            } else if (onClick) {
+              await onClick();
+            }
+          } catch (error) {
+            setLoading(false);
+            throw error;
+          }
+          setLoading(false);
+        },
+        500,
+        {leading: true}
+      )}
+    >
+      <View>
+        {Boolean(loading) ? (
+          <View>
+            <ActivityIndicator color={color as any} size="small" />
+          </View>
+        ) : (
+          <FontAwesome6 brand="solid" color={color as any} name={iconName} size={16} />
```

Comment:
I added this to Jo's question list because I have noticed that I've been having to hard code my icons since the styles don't match any sizes we currently have in the theme

Reply:
Using an icon within a component is not equal to the iconComponent. Meaning the scenarios we will use icon component are not equal to using iconButton. 
There will be instances where we will need to use react-native components since we are building the base components to be used together to make the app. We are not always using base components to build base components. If we are it is labeled as such in Figma